ddtrace/contrib/starlette/patch.py

```diff
@@ -65,3 +58,36 @@ def unpatch():
     setattr(starlette, "_datadog_patch", False)
 
     _u(starlette.applications.Starlette, "__init__")
+
+    handlers = [
+        starlette.routing.Mount,
+        starlette.routing.Route,
+        starlette.routing.Host,
+        starlette.routing.WebSocketRoute,
+    ]
+
+    # Unwrap all of the handlers
+    for handler in handlers:
+        _u(handler, "handle")
+
+
+def traced_handler(wrapped, instance, args, kwargs):
+    def _wrap(scope, receive, send):
+        # Since handle can be called multiple times for one request, we take the path of each instance
+        # Then combine them at the end to get the correct resource name
+        if isinstance(instance, BaseRoute):
+            if "__dd_paths__" in scope:
+                scope["__dd_paths__"].append(instance.path)
+
+            else:
+                scope["__dd_paths__"] = [instance.path]
+
+            method = scope["method"] if "method" in scope else ""
```

Comment:
do we have cases where we might not have `method`?

also, we might want a different formatting case below if we don't have method. as-is if `method = ""`, then we'll get a resource name like `" /path/with/space/in/front"`

Reply:
I didn't think so, but then tests were failing once I made the PR about no method existing, so I added the check. 

Ok, do you think we'd want no space in front?

Reply:
got it, yeah, if there is not method we shouldn't have the blank space in front, or should we default to an `UNKNOWN` or something....? (I am actually not sure here)

Reply:
It's currently setup so that if there's no method, we just use `path` as the resource.